cartographer/mapping_3d/pose_graph.cc

```diff
@@ -338,6 +338,23 @@ void PoseGraph::HandleWorkQueue() {
           trimmer->Trim(&trimming_handle);
         }
 
+        {
+          auto GetTrimmerToRemove = [&]() {
```

Comment:
Can't you use [std::remove_if](http://en.cppreference.com/w/cpp/algorithm/remove) here?